snuba_sdk/legacy.py

```diff
@@ -18,48 +18,50 @@ def parse_datetime(date_str: str) -> datetime:
     return datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S.%f")
 
 
-def json_to_snql(body: Mapping[str, Any], entity: str) -> Query:
-    def to_exp(value: Union[str, List[Any]]) -> Union[Column, Function]:
-        if not isinstance(value, list):
-            return Column(value)
-
-        children = []
-        if isinstance(value[1], list):
-            children = list(map(to_exp, value[1]))
-        elif value[1]:
-            children = [to_exp(value[1])]
-
-        alias = value[2] if len(value) > 2 else None
-        return Function(value[0], children, alias)
-
-    def to_scalar(value: Any) -> Any:
-        if isinstance(value, list):
-            return list(map(to_scalar, value))
-        elif isinstance(value, tuple):
-            return tuple(map(to_scalar, value))
-
-        if isinstance(value, str):
-            try:
-                date_scalar = parse_datetime(value)
-                return date_scalar
-            except ValueError:
-                return value
-
-        return value
+def parse_scalar(value: Any) -> Any:
+    if isinstance(value, list):
+        return list(map(parse_scalar, value))
+    elif isinstance(value, tuple):
+        return tuple(map(parse_scalar, value))
+
+    if isinstance(value, str):
+        try:
+            date_scalar = parse_datetime(value)
+            return date_scalar
+        except ValueError:
+            return value
+
+    return value
+
+
+def parse_exp(value: Any) -> Any:
+    if isinstance(value, str):
+        return Column(value)
+    if not isinstance(value, list):
+        return parse_scalar(value)
+
+    alias = value[2] if len(value) > 2 else None
 
+    if value[0].endswith("()") and not value[1]:
+        return Function(value[0].strip("()"), [], alias)
```

Comment:
What is this case for ?

Reply:
Sometimes clients are sending `["count()", None, ...]`.